src/components/OverridableAsset.vue

```diff
@@ -8,7 +8,7 @@
   See https://swift.org/CONTRIBUTORS.txt for Swift project authors
 -->
 <template>
-  <ImageAsset v-if="shouldUseAsset" :variants="variants" />
+  <ImageAsset v-if="shouldUseAsset" :variants="variants" :loading="null" />
```

Comment:
This change didn't have a noticeable impact on the rendering for me—I'm still seeing the visual flicker when scrolling the sidebar quickly when there are many items with custom icons.

Looking closer at this, I think that the actual flicker/shift happens when the logic for calculating the optimal width for the `<img>` tag happens here: https://github.com/apple/swift-docc-render/blob/881b0a551e544384f69690dfd26b9abcd457ecb8/src/components/ImageAsset.vue#L179

Since the design of the sidebar dictates the size of the icons for this particular part of the UI, I'm wondering if this could be fixed by simply enforcing a fixed width for overwritten icons (using the img tag) in the sidebar—that should prevent that logic from needing to happen asynchronously and fix the shifting issue maybe?

Reply:
I saw two issues, one is flickering, the other is shifting of size.

Once I added that `flex: 1` to the `picture` it fixed the size shifting and the flicker got removed from the `loading=null` thing.

I will do more testing today.

Reply:
The shifting of size issue has been fixed with the latest change to add the `key`.

I'm still seeing an initial flicker as each distinct icon loads when its parent sidebar item is disclosed. Do you think there's anything else we can do to minimize that?

I'm still curious if maybe there's an easy way to avoid calculating the sizes of each image when we already know that they're going to be a certain size in this part of the UI. Could that calculation be causing the first load flicker or is it unavoidable without preloading the images? Looking closer, I'm actually seeing an error for that logic for each icon—maybe it doesn't even work well with SVG assets...